wtfsic/src/components/AddRecipeForm.jsx

```diff
@@ -15,23 +15,43 @@ const postData = async (url = '', data = {}) => {
 
 // Those hard coded values are only for testing purposes
 const cleanRecipeData = (formData) => {
-     // TODO : check Django session to get user creds
+    // TODO : check Django session to get user creds
     // https://stackoverflow.com/questions/59524305/how-to-access-django-session-in-react
-        formData.author = 1
-        formData.ingredients = []
-        formData.cook_time = '00:30:00'
-        formData.prep_time = '00:10:00'
+    formData.author = 1
+    formData.cook_time = '00:30:00'
+    formData.prep_time = '00:10:00'
     return formData
 }
 
- const addRecipe = async (data) => {
+const addRecipe = async (data) => {
     const cleanData = cleanRecipeData(data)
-     return await postData('/api/recipes', cleanData)
+    return await postData('/api/recipes', cleanData)
 }
 
-
 const AddRecipeForm = () => {
-    const {register, handleSubmit } = useForm();
+    const { register, control, handleSubmit } = useForm();
+    const { fields, append, remove } = useFieldArray({
+        control,
+        name: 'ingredients'
+    });
+
+
+    const [units, setUnits] = useState()
+    useEffect(() => {
+        const fetchUnits = async () => {
+          try {
+            const apiUrl = '/api/ingredients';
+            // https://www.django-rest-framework.org/api-guide/metadata/#metadata
+            const response = await fetch(apiUrl, { method: 'OPTIONS' });
+            const { actions } = await response.json();
```

Comment:
[question/non-blocking] Do you need to `await` the response here in order to get `{ actions }` ?